xilem/examples/stopwatch.rs

```diff
@@ -0,0 +1,261 @@
+    fork(
+        flex((
+            FlexSpacer::Fixed(5.0),
+            label(get_formatted_duration(&data.displayed_duration)).text_size(70.0),
+            flex((lap_reset_button(data), start_stop_button(data))).direction(Axis::Horizontal),
+            FlexSpacer::Fixed(1.0),
+            laps_section(data),
+            label(data.displayed_error.as_ref()),
+        )),
+        data.active.then(|| {
+            // Only update while active.
+            async_repeat(
+                |proxy| async move {
+                    let mut interval = time::interval(Duration::from_millis(50));
+                    loop {
+                        interval.tick().await;
+                        let Ok(()) = proxy.message(()) else {
+                            break;
+                        };
+                    }
+                },
+                |data: &mut Stopwatch, ()| {
+                    data.update_display();
+                },
+            )
+        }),
+    )
+}
+
+/// Creates a list of items that shows the lap number, split time, and total cumulative time.
+fn laps_section(data: &mut Stopwatch) -> Vec<AnyFlexChild<Stopwatch>> {
+    let mut items: Vec<AnyFlexChild<Stopwatch>> = Vec::new();
+    let mut total_dur = Duration::ZERO;
+    let current_lap = data.completed_lap_splits.len();
+
+    for (i, split_dur) in data.completed_lap_splits.iter().enumerate() {
+        total_dur = total_dur.add(*split_dur);
+        items.push(single_lap(i, split_dur, &total_dur).into_any_flex());
+    }
+    let current_split_duration = data.get_current_duration().sub(total_dur);
+    let mut reversed = Vec::new();
+    reversed.push(
+        single_lap(
+            current_lap,
+            &current_split_duration,
+            &data.get_current_duration(),
+        )
+        .into_any_flex(),
+    );
+    for item in items.into_iter().rev() {
+        reversed.push(item);
+    }
+    reversed
+}
+
+fn single_lap(
+    lap_id: usize,
+    split_dur: &Duration,
+    total_dur: &Duration,
+) -> impl WidgetView<Stopwatch> {
+    flex((
+        FlexSpacer::Flex(1.0),
+        label(format!("Lap {}", lap_id + 1)),
+        label(get_formatted_duration(split_dur)),
+        label(get_formatted_duration(total_dur)),
+        FlexSpacer::Flex(1.0),
+    ))
+    .direction(Axis::Horizontal)
+    .cross_axis_alignment(CrossAxisAlignment::Center)
+    .main_axis_alignment(MainAxisAlignment::Start)
+    .must_fill_major_axis(true)
+}
+
+fn start_stop_button(data: &mut Stopwatch) -> impl WidgetView<Stopwatch> {
+    if data.active {
+        Either::A(button("Stop", |data: &mut Stopwatch| {
+            data.stop();
+        }))
+    } else {
+        Either::B(button("Start", |data: &mut Stopwatch| {
+            data.start();
+        }))
+    }
+}
+
+fn lap_reset_button(data: &mut Stopwatch) -> impl WidgetView<Stopwatch> {
+    if data.active {
+        Either::A(button("  Lap  ", |data: &mut Stopwatch| {
```

Comment:
:laughing:

Reply:
Didn't need to use a sized_box because of that 😁